[BUG] PHP 8.2 gives me build errors

### Bug description
PHP 8.2 gives me build errors

```
$ apk --no-cache add cargo curl curl-dev findutils gcc make musl-dev php82-dev php82-openssl php82-pear
$ pecl82 install -f datadog_trace-0.88.1
  ...
   Compiling ddtrace-php v0.0.1 (/tmp/pear/temp/datadog_trace/components-rs)
   Compiling ddtelemetry-ffi v2.2.0 (/tmp/pear/temp/datadog_trace/libdatadog/ddtelemetry-ffi)
error: failed to run custom build command for `ddtrace-php v0.0.1 (/tmp/pear/temp/datadog_trace/components-rs)`

Caused by:
  process didn't exit successfully: `/tmp/pear/temp/pear-build-defaultuserEoIjgc/datadog_trace-0.88.1/target/debug/build/ddtrace-php-ca5bc27c59f8d437/build-script-build` (exit status: 1)
  --- stderr
  Failed generating mock_php.c: Could not read : No such file or directory (os error 2)
warning: build failed, waiting for other jobs to finish...
strip: '/tmp/pear/temp/pear-build-defaultuserEoIjgc/datadog_trace-0.88.1/target/debug/libddtrace_php.a': No such file
make: *** [Makefile:211: /tmp/pear/temp/pear-build-defaultuserEoIjgc/datadog_trace-0.88.1/target/debug/libddtrace_php.a] Error 1
ERROR: `make' failed
```

### PHP version
```
PHP 8.2.7 (cli) (built: Jun  9 2023 00:27:17) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.7, Copyright (c) Zend Technologies
```
### Tracer version
- 0.88.1
- 0.88.0

### Installed extensions
```
[PHP Modules]
Core
date
filter
hash
json
libxml
openssl
pcre
random
readline
Reflection
SPL
standard
xml
zlib

[Zend Modules]
```

### OS info
```
NAME="Alpine Linux"
VERSION_ID=3.18.2
PRETTY_NAME="Alpine Linux v3.18"
```

--- a/.circleci/continue_config.yml
+++ b/.circleci/continue_config.yml
@@ -964,9 +964,6 @@ jobs:
       - store_artifacts:
           path: /tmp/artifacts/
       - <<: *STEP_COMPOSER_CACHE_SAVE
-      - persist_to_workspace:
-          root: .
-          paths: [tmp/build_extension]
       - <<: *STEP_STORE_TEST_RESULTS
 
   integration:

--- a/config.m4
+++ b/config.m4
@@ -268,7 +268,7 @@ if test "$PHP_DDTRACE" != "no"; then
   if test "$ext_shared" = "yes"; then
     all_object_files=$(for src in $DD_TRACE_PHP_SOURCES $ZAI_SOURCES; do printf ' %s' "${src%?}lo"; done)
     all_object_files_newlines=$(for src in $DD_TRACE_PHP_SOURCES $ZAI_SOURCES; do printf '\\n$(builddir)/%s' "$(dirname "$src")/$objdir/$(basename "${src%?}o")"; done)
-    php_binary=$(php-config --php-binary)
+    php_binary=$($PHP_CONFIG --php-binary)
     ddtrace_mock_sources='DD_SIDECAR_MOCK_SOURCES="$$(printf "'"$php_binary$all_object_files_newlines"'")"'
   else
     all_object_files=

